CONTRIBUTING.md

```diff
@@ -0,0 +1,192 @@
+   ```
+
+2. Install dependencies with your preferred package manager:
+
+   ```bash
+   npm install
+   ```
+
+3. Create a new topic branch (off the main project development branch) to contain your feature, change, or fix:
+
+   ```bash
+   git checkout -b <topic-branch-name>
+   ```
+
+4. Follow our [coding conventions](#coding-conventions) while developing your changes. You can also run `npm run lint` to match our requirements.
+
+5. Write clear and meaningful git commit messages. Please follow our [commit message conventions](#commit-conventions) while committing to your branch.
+
+6. Make sure to update or add to the tests when appropriate. Run the appropriate testing suites to check that all tests pass after you've made changes. You can read about our different types of tests in our [Testing](#testing) section.
+
+7. If you added or changed a feature, make sure to document it accordingly in the [README.md](https://github.com/openpgpjs/openpgpjs/blob/main/README.md) file.
+
+## Coding conventions
+
+We ensure code consistency through our [ESLint config](https://github.com/openpgpjs/openpgpjs/blob/main/.eslintrc.js). You can run `npm run lint` to lint your changes. When in doubt, follow the style of the surrounding code.
+
+## Commit conventions
+
+We follow the seven rules of a great Git commit message, which is extensively described in [this blog post](https://cbea.ms/git-commit/), but in short the principles are:
+
+1. Separate subject from body with a blank line
+2. Limit the subject line to 50 characters
+3. Capitalize the subject line
+4. Do not end the subject line with a period
+5. Use the imperative mood in the subject line
+6. Wrap the body at 72 characters
+7. Use the body to explain what and why vs. how
+
+You can also take a look at our [main branch commits](https://github.com/openpgpjs/openpgpjs/commits/main).
+
+## Testing
+
+We have multiple types of tests. They are located in the root [`/test`](https://github.com/openpgpjs/openpgpjs/tree/main/test) folder.
+Before you can run the tests you need to build the project using `npm run build` and this will bundle the project and the tests for it be used in our different testing environments.
+
+**Unit Tests:**
+Unit tests are handled by mocha and run inside node, you can run them using this command:
+
+```sh
+npm run test
+```
+
+**Browser Tests:**
+For debugging browser errors, you can run `npm start` and open [`http://localhost:8080/test/unittests.html`](http://localhost:8080/test/unittests.html) in a browser, or run one the following commands:
+
+This will run the tests inside a browser environment:
+
+```sh
+   npm run browsertest
+```
+
+You can run the tests for multiple headless browsers using KarmaJs:
+
+` npm run test-browser`
+
+If you wanna start the tests using KarmaJs and BrowserStack:
+`npm run test-browserstack`
+
+**_NOTE_**: You need to configure you env with `BROWSER_STACK_ACCESS_KEY` and `BROWSERSTACK_USERNAME` to be able to run the BrowserStack tests.
```

Comment:
I would spell out the various options tbh, it's good to have a reference.

Running the tests with Karma (`npm run test-browser`) is considerably faster than using `browsertest`, and automatically targets multiple browsers.
Similarly, it can be useful to trigger the BrowserStack tests locally for debugging -- I wouldn't implicitly encourage using the CI for that.